teamcity: failed tests on master: test/TestStoreRangeMergeReadoptedBothFollowers, testrace/TestStoreRangeMergeReadoptedBothFollowers

The following tests appear to have failed:

[#862073](https://teamcity.cockroachdb.com/viewLog.html?buildId=862073):


```
--- FAIL: test/TestStoreRangeMergeReadoptedBothFollowers (0.340s)
client_merge_test.go:1783: expected "cannot apply snapshot: snapshot intersects existing range" error, but got snapshot failed: (n3,s3):?: remote failed to apply snapshot for reason failed to apply snapshot: snapshot intersects existing range [n3,s3,r2/3:{b-/Max}]
------- Stdout: -------
I180826 22:50:38.966349 55748 storage/store.go:1541  [s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180826 22:50:38.968382 55748 gossip/gossip.go:382  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:46319" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180826 22:50:38.974659 55748 storage/store.go:1541  [s2] [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
W180826 22:50:38.976410 55748 gossip/gossip.go:1371  [n2] no incoming or outgoing connections
I180826 22:50:38.976707 56148 gossip/client.go:129  [n2] started gossip client to 127.0.0.1:46319
I180826 22:50:38.984086 55748 gossip/gossip.go:382  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:35531" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180826 22:50:38.991308 55748 storage/store.go:1541  [s3] [n3,s3]: failed initial metrics computation: [n3,s3]: system config not yet available
W180826 22:50:39.002953 55748 gossip/gossip.go:1371  [n3] no incoming or outgoing connections
I180826 22:50:39.003048 55748 gossip/gossip.go:382  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:35541" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180826 22:50:39.004148 56285 gossip/client.go:129  [n3] started gossip client to 127.0.0.1:46319
I180826 22:50:39.061898 55748 rpc/nodedialer/nodedialer.go:92  [s1,r1/1:/M{in-ax}] connection to n2 established
I180826 22:50:39.067689 55748 storage/store_snapshot.go:615  [s1,r1/1:/M{in-ax}] sending preemptive snapshot 6b258572 at applied index 16
I180826 22:50:39.068018 55748 storage/store_snapshot.go:657  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):?: kv pairs: 47, log entries: 6, rate-limit: 2.0 MiB/sec, 6ms
I180826 22:50:39.068448 56267 storage/replica_raftstorage.go:784  [s2,r1/?:{-}] applying preemptive snapshot at index 16 (id=6b258572, encoded size=8212, 1 rocksdb batches, 6 log entries)
I180826 22:50:39.069482 56267 storage/replica_raftstorage.go:790  [s2,r1/?:/M{in-ax}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I180826 22:50:39.069876 55748 storage/replica_command.go:803  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I180826 22:50:39.071558 55748 storage/replica.go:3743  [txn=74f4e73d,s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I180826 22:50:39.072227 55748 rpc/nodedialer/nodedialer.go:92  [s1,r1/1:/M{in-ax}] connection to n3 established
I180826 22:50:39.073154 55748 storage/store_snapshot.go:615  [s1,r1/1:/M{in-ax}] sending preemptive snapshot ef66238f at applied index 18
I180826 22:50:39.083721 55748 storage/store_snapshot.go:657  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):?: kv pairs: 50, log entries: 8, rate-limit: 2.0 MiB/sec, 11ms
I180826 22:50:39.084313 56318 storage/replica_raftstorage.go:784  [s3,r1/?:{-}] applying preemptive snapshot at index 18 (id=ef66238f, encoded size=9154, 1 rocksdb batches, 8 log entries)
I180826 22:50:39.085811 56318 storage/replica_raftstorage.go:790  [s3,r1/?:/M{in-ax}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=1ms commit=1ms]
I180826 22:50:39.086542 55748 storage/replica_command.go:803  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=0]
I180826 22:50:39.090590 56097 rpc/nodedialer/nodedialer.go:92  connection to n1 established
I180826 22:50:39.092854 55748 storage/replica.go:3743  [txn=5b9debd3,s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I180826 22:50:39.234591 55748 storage/replica_command.go:289  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "b" [r2]
I180826 22:50:39.249304 55748 storage/replica_command.go:803  [s1,r1/1:{/Min-b}] change replicas (REMOVE_REPLICA (n3,s3):3): read existing descriptor r1:{/Min-b} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=1]
I180826 22:50:39.251875 55748 storage/replica.go:3743  [txn=537f8b09,s1,r1/1:{/Min-b}] proposing REMOVE_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2] next=4
I180826 22:50:39.253576 55748 storage/replica_command.go:803  [s1,r2/1:{b-/Max}] change replicas (REMOVE_REPLICA (n3,s3):3): read existing descriptor r2:{b-/Max} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=0]
I180826 22:50:39.259002 55748 storage/replica.go:3743  [txn=382cd671,s1,r2/1:{b-/Max}] proposing REMOVE_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2] next=4
I180826 22:50:39.260622 55748 storage/replica_command.go:421  [s1,r1/1:{/Min-b}] initiating a merge of r2:{b-/Max} [(n1,s1):1, (n2,s2):2, next=4, gen=0] into this range
I180826 22:50:39.270006 56000 storage/store.go:2656  [txn=e8868028,s1,r1/1:{/Min-b}] removing replica r2/1
I180826 22:50:39.271094 56086 storage/store.go:2656  [s2,r1/2:{/Min-b}] removing replica r2/2
I180826 22:50:39.271436 55748 storage/store_snapshot.go:615  sending preemptive snapshot db81f2a3 at applied index 30
I180826 22:50:39.271809 55748 storage/store_snapshot.go:657  streamed snapshot to (n3,s3):?: kv pairs: 69, log entries: 20, rate-limit: 2.0 MiB/sec, 1ms
E180826 22:50:39.273702 55938 storage/store.go:3765  [s3,r2/3:{b-/Max}] unable to add to replica GC queue: queue disabled
E180826 22:50:39.274834 56210 storage/replica_proposal.go:730  [s3,r1/3:{/Min-b}] unable to add to replica GC queue: queue disabled
E180826 22:50:39.275679 55938 storage/store.go:3746  [s3,r1/3:{/Min-b}] unable to add to replica GC queue: queue disabled
E180826 22:50:39.275734 55938 storage/store.go:3746  [s3,r1/3:{/Min-b}] unable to add to replica GC queue: queue disabled
E180826 22:50:39.275757 55938 storage/store.go:3746  [s3,r1/3:{/Min-b}] unable to add to replica GC queue: queue disabled
I180826 22:50:39.276096 56141 storage/store.go:3508  [s3,r1/3:{/Min-b}] cannot apply snapshot: snapshot intersects existing range [n3,s3,r2/3:{b-/Max}]
I180826 22:50:39.276542 56371 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] wait-for-merge
W180826 22:50:39.278119 56341 storage/raft_transport.go:583  while processing outgoing Raft queue to node 3: rpc error: code = Unavailable desc = transport is closing:
W180826 22:50:39.278468 56288 gossip/infostore.go:302  [n3] node unavailable; try another peer
W180826 22:50:39.278527 56288 gossip/infostore.go:302  [n3] node unavailable; try another peer
--- FAIL: testrace/TestStoreRangeMergeReadoptedBothFollowers (1.400s)

------- Stdout: -------
I180826 23:05:16.251855 55967 storage/store.go:1541  [s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180826 23:05:16.265667 55967 gossip/gossip.go:382  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:41941" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180826 23:05:16.314071 55967 storage/store.go:1541  [s2] [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
W180826 23:05:16.330868 55967 gossip/gossip.go:1371  [n2] no incoming or outgoing connections
I180826 23:05:16.331944 55967 gossip/gossip.go:382  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:38089" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180826 23:05:16.334505 56282 gossip/client.go:129  [n2] started gossip client to 127.0.0.1:41941
I180826 23:05:16.452274 55967 storage/store.go:1541  [s3] [n3,s3]: failed initial metrics computation: [n3,s3]: system config not yet available
W180826 23:05:16.461088 55967 gossip/gossip.go:1371  [n3] no incoming or outgoing connections
I180826 23:05:16.462122 55967 gossip/gossip.go:382  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:34319" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180826 23:05:16.465516 55999 gossip/client.go:129  [n3] started gossip client to 127.0.0.1:41941
I180826 23:05:16.507726 55967 rpc/nodedialer/nodedialer.go:92  [s1,r1/1:/M{in-ax}] connection to n2 established
I180826 23:05:16.511627 55967 storage/store_snapshot.go:615  [s1,r1/1:/M{in-ax}] sending preemptive snapshot 18992802 at applied index 16
I180826 23:05:16.513028 55967 storage/store_snapshot.go:657  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):?: kv pairs: 47, log entries: 6, rate-limit: 2.0 MiB/sec, 4ms
I180826 23:05:16.515479 56438 storage/replica_raftstorage.go:784  [s2,r1/?:{-}] applying preemptive snapshot at index 16 (id=18992802, encoded size=8216, 1 rocksdb batches, 6 log entries)
I180826 23:05:16.518754 56438 storage/replica_raftstorage.go:790  [s2,r1/?:/M{in-ax}] applied preemptive snapshot in 3ms [clear=0ms batch=0ms entries=2ms commit=0ms]
I180826 23:05:16.522378 55967 storage/replica_command.go:803  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I180826 23:05:16.533305 55967 storage/replica.go:3743  [txn=543952db,s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I180826 23:05:16.549149 55967 rpc/nodedialer/nodedialer.go:92  [s1,r1/1:/M{in-ax}] connection to n3 established
I180826 23:05:16.553989 55967 storage/store_snapshot.go:615  [s1,r1/1:/M{in-ax}] sending preemptive snapshot c9b358d9 at applied index 18
I180826 23:05:16.559795 55967 storage/store_snapshot.go:657  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):?: kv pairs: 50, log entries: 8, rate-limit: 2.0 MiB/sec, 8ms
I180826 23:05:16.562088 56251 storage/replica_raftstorage.go:784  [s3,r1/?:{-}] applying preemptive snapshot at index 18 (id=c9b358d9, encoded size=9162, 1 rocksdb batches, 8 log entries)
I180826 23:05:16.563517 56226 rpc/nodedialer/nodedialer.go:92  connection to n1 established
I180826 23:05:16.566169 56251 storage/replica_raftstorage.go:790  [s3,r1/?:/M{in-ax}] applied preemptive snapshot in 4ms [clear=0ms batch=0ms entries=3ms commit=0ms]
I180826 23:05:16.569652 55967 storage/replica_command.go:803  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=0]
I180826 23:05:16.589947 55967 storage/replica.go:3743  [txn=8b017cf6,s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I180826 23:05:16.758405 55967 storage/replica_command.go:289  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "b" [r2]
I180826 23:05:16.803686 55967 storage/replica_command.go:803  [s1,r1/1:{/Min-b}] change replicas (REMOVE_REPLICA (n3,s3):3): read existing descriptor r1:{/Min-b} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=1]
I180826 23:05:16.827254 55967 storage/replica.go:3743  [txn=d1da0838,s1,r1/1:{/Min-b}] proposing REMOVE_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2] next=4
E180826 23:05:16.838270 56121 storage/replica_proposal.go:730  [s3,r1/3:{/Min-b}] unable to add to replica GC queue: queue disabled
I180826 23:05:16.841235 55967 storage/replica_command.go:803  [s1,r2/1:{b-/Max}] change replicas (REMOVE_REPLICA (n3,s3):3): read existing descriptor r2:{b-/Max} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=0]
E180826 23:05:16.845985 56467 storage/store.go:3746  [s3,r1/3:{/Min-b}] unable to add to replica GC queue: queue disabled
I180826 23:05:16.907152 55967 storage/replica.go:3743  [txn=f511d453,s1,r2/1:{b-/Max}] proposing REMOVE_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2] next=4
E180826 23:05:16.930113 56342 storage/replica_proposal.go:730  [s3,r2/3:{b-/Max}] unable to add to replica GC queue: queue disabled
E180826 23:05:16.932499 56467 storage/store.go:3746  [s3,r2/3:{b-/Max}] unable to add to replica GC queue: queue disabled
I180826 23:05:16.934090 55967 storage/replica_command.go:421  [s1,r1/1:{/Min-b}] initiating a merge of r2:{b-/Max} [(n1,s1):1, (n2,s2):2, next=4, gen=0] into this range
I180826 23:05:16.991807 56077 storage/store.go:2656  [txn=a6f2686a,s1,r1/1:{/Min-b}] removing replica r2/1
I180826 23:05:16.993809 56212 storage/store.go:2656  [s2,r1/2:{/Min-b}] removing replica r2/2
I180826 23:05:17.003553 55967 storage/store.go:2656  [s3,r1/3:{/Min-b}] removing replica r1/3
I180826 23:05:17.004791 55967 storage/replica.go:880  [s3,r1/3:{/Min-b}] removed 14 (9+5) keys in 1ms [clear=0ms commit=0ms]
I180826 23:05:17.015314 55967 storage/store.go:2656  [s3,r2/3:{b-/Max}] removing replica r2/3
I180826 23:05:17.017047 55967 storage/replica.go:880  [s3,r2/3:{b-/Max}] removed 40 (34+6) keys in 1ms [clear=1ms commit=0ms]
I180826 23:05:17.021474 55967 storage/store_snapshot.go:615  sending preemptive snapshot f906e071 at applied index 33
I180826 23:05:17.023420 55967 storage/store_snapshot.go:657  streamed snapshot to (n3,s3):?: kv pairs: 72, log entries: 23, rate-limit: 2.0 MiB/sec, 4ms
I180826 23:05:17.025422 56510 storage/replica_raftstorage.go:784  [s3,r1/?:{-}] applying preemptive snapshot at index 33 (id=f906e071, encoded size=15355, 1 rocksdb batches, 23 log entries)
I180826 23:05:17.044485 56510 storage/replica_raftstorage.go:790  [s3,r1/?:/M{in-ax}] applied preemptive snapshot in 19ms [clear=0ms batch=0ms entries=17ms commit=1ms]
I180826 23:05:17.047868 55967 storage/replica_command.go:803  change replicas (ADD_REPLICA (n3,s3):4): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=4, gen=2]
I180826 23:05:17.062764 55967 storage/replica.go:3743  [txn=71ffb5c7,s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n3,s3):4): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):4] next=5
I180826 23:05:17.092682 55967 storage/replica_range_lease.go:554  transferring lease to s3
I180826 23:05:17.116185 55967 storage/replica_range_lease.go:617  done transferring lease to s3: <nil>client_merge_test.go:1783: expected "cannot apply snapshot: snapshot intersects existing range" error, but got snapshot failed: (n3,s3):?: remote failed to apply snapshot for reason failed to apply snapshot: snapshot intersects existing range [n3,s3,r2/3:{b-/Max}]
------- Stdout: -------
I180826 22:50:38.966349 55748 storage/store.go:1541  [s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180826 22:50:38.968382 55748 gossip/gossip.go:382  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:46319" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180826 22:50:38.974659 55748 storage/store.go:1541  [s2] [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
W180826 22:50:38.976410 55748 gossip/gossip.go:1371  [n2] no incoming or outgoing connections
I180826 22:50:38.976707 56148 gossip/client.go:129  [n2] started gossip client to 127.0.0.1:46319
I180826 22:50:38.984086 55748 gossip/gossip.go:382  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:35531" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180826 22:50:38.991308 55748 storage/store.go:1541  [s3] [n3,s3]: failed initial metrics computation: [n3,s3]: system config not yet available
W180826 22:50:39.002953 55748 gossip/gossip.go:1371  [n3] no incoming or outgoing connections
I180826 22:50:39.003048 55748 gossip/gossip.go:382  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:35541" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180826 22:50:39.004148 56285 gossip/client.go:129  [n3] started gossip client to 127.0.0.1:46319
I180826 22:50:39.061898 55748 rpc/nodedialer/nodedialer.go:92  [s1,r1/1:/M{in-ax}] connection to n2 established
I180826 22:50:39.067689 55748 storage/store_snapshot.go:615  [s1,r1/1:/M{in-ax}] sending preemptive snapshot 6b258572 at applied index 16
I180826 22:50:39.068018 55748 storage/store_snapshot.go:657  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):?: kv pairs: 47, log entries: 6, rate-limit: 2.0 MiB/sec, 6ms
I180826 22:50:39.068448 56267 storage/replica_raftstorage.go:784  [s2,r1/?:{-}] applying preemptive snapshot at index 16 (id=6b258572, encoded size=8212, 1 rocksdb batches, 6 log entries)
I180826 22:50:39.069482 56267 storage/replica_raftstorage.go:790  [s2,r1/?:/M{in-ax}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I180826 22:50:39.069876 55748 storage/replica_command.go:803  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I180826 22:50:39.071558 55748 storage/replica.go:3743  [txn=74f4e73d,s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I180826 22:50:39.072227 55748 rpc/nodedialer/nodedialer.go:92  [s1,r1/1:/M{in-ax}] connection to n3 established
I180826 22:50:39.073154 55748 storage/store_snapshot.go:615  [s1,r1/1:/M{in-ax}] sending preemptive snapshot ef66238f at applied index 18
I180826 22:50:39.083721 55748 storage/store_snapshot.go:657  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):?: kv pairs: 50, log entries: 8, rate-limit: 2.0 MiB/sec, 11ms
I180826 22:50:39.084313 56318 storage/replica_raftstorage.go:784  [s3,r1/?:{-}] applying preemptive snapshot at index 18 (id=ef66238f, encoded size=9154, 1 rocksdb batches, 8 log entries)
I180826 22:50:39.085811 56318 storage/replica_raftstorage.go:790  [s3,r1/?:/M{in-ax}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=1ms commit=1ms]
I180826 22:50:39.086542 55748 storage/replica_command.go:803  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=0]
I180826 22:50:39.090590 56097 rpc/nodedialer/nodedialer.go:92  connection to n1 established
I180826 22:50:39.092854 55748 storage/replica.go:3743  [txn=5b9debd3,s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I180826 22:50:39.234591 55748 storage/replica_command.go:289  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "b" [r2]
I180826 22:50:39.249304 55748 storage/replica_command.go:803  [s1,r1/1:{/Min-b}] change replicas (REMOVE_REPLICA (n3,s3):3): read existing descriptor r1:{/Min-b} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=1]
I180826 22:50:39.251875 55748 storage/replica.go:3743  [txn=537f8b09,s1,r1/1:{/Min-b}] proposing REMOVE_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2] next=4
I180826 22:50:39.253576 55748 storage/replica_command.go:803  [s1,r2/1:{b-/Max}] change replicas (REMOVE_REPLICA (n3,s3):3): read existing descriptor r2:{b-/Max} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=0]
I180826 22:50:39.259002 55748 storage/replica.go:3743  [txn=382cd671,s1,r2/1:{b-/Max}] proposing REMOVE_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2] next=4
I180826 22:50:39.260622 55748 storage/replica_command.go:421  [s1,r1/1:{/Min-b}] initiating a merge of r2:{b-/Max} [(n1,s1):1, (n2,s2):2, next=4, gen=0] into this range
I180826 22:50:39.270006 56000 storage/store.go:2656  [txn=e8868028,s1,r1/1:{/Min-b}] removing replica r2/1
I180826 22:50:39.271094 56086 storage/store.go:2656  [s2,r1/2:{/Min-b}] removing replica r2/2
I180826 22:50:39.271436 55748 storage/store_snapshot.go:615  sending preemptive snapshot db81f2a3 at applied index 30
I180826 22:50:39.271809 55748 storage/store_snapshot.go:657  streamed snapshot to (n3,s3):?: kv pairs: 69, log entries: 20, rate-limit: 2.0 MiB/sec, 1ms
E180826 22:50:39.273702 55938 storage/store.go:3765  [s3,r2/3:{b-/Max}] unable to add to replica GC queue: queue disabled
E180826 22:50:39.274834 56210 storage/replica_proposal.go:730  [s3,r1/3:{/Min-b}] unable to add to replica GC queue: queue disabled
E180826 22:50:39.275679 55938 storage/store.go:3746  [s3,r1/3:{/Min-b}] unable to add to replica GC queue: queue disabled
E180826 22:50:39.275734 55938 storage/store.go:3746  [s3,r1/3:{/Min-b}] unable to add to replica GC queue: queue disabled
E180826 22:50:39.275757 55938 storage/store.go:3746  [s3,r1/3:{/Min-b}] unable to add to replica GC queue: queue disabled
I180826 22:50:39.276096 56141 storage/store.go:3508  [s3,r1/3:{/Min-b}] cannot apply snapshot: snapshot intersects existing range [n3,s3,r2/3:{b-/Max}]
I180826 22:50:39.276542 56371 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] wait-for-merge
W180826 22:50:39.278119 56341 storage/raft_transport.go:583  while processing outgoing Raft queue to node 3: rpc error: code = Unavailable desc = transport is closing:
W180826 22:50:39.278468 56288 gossip/infostore.go:302  [n3] node unavailable; try another peer
W180826 22:50:39.278527 56288 gossip/infostore.go:302  [n3] node unavailable; try another peer

```

Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/client_merge_test.go
+++ b/pkg/storage/client_merge_test.go
@@ -1727,6 +1727,7 @@ func TestStoreRangeMergeReadoptedBothFollowers(t *testing.T) {
 	mtc.Start(t, 3)
 	defer mtc.Stop()
 	store0, store2 := mtc.Store(0), mtc.Store(2)
+	distSender := mtc.distSenders[0]
 
 	// Create two ranges on all nodes.
 	mtc.replicateRange(roachpb.RangeID(1), 1, 2)
@@ -1735,16 +1736,26 @@ func TestStoreRangeMergeReadoptedBothFollowers(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// Wait for store2 to hear about the split.
-	var lhsRepl2, rhsRepl2 *storage.Replica
-	testutils.SucceedsSoon(t, func() error {
-		lhsRepl2, err = store2.GetReplica(lhsDesc.RangeID)
-		if err != nil {
-			return err
+	// Wait for all stores to have fully processed the split.
+	for _, key := range []roachpb.Key{roachpb.Key("a"), roachpb.Key("b")} {
+		if _, pErr := client.SendWrapped(ctx, distSender, incrementArgs(key, 1)); pErr != nil {
+			t.Fatal(pErr)
 		}
-		rhsRepl2, err = store2.GetReplica(rhsDesc.RangeID)
-		return err
-	})
+		mtc.waitForValues(key, []int64{1, 1, 1})
+	}
+
+	lhsRepl0, err := store0.GetReplica(lhsDesc.RangeID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	lhsRepl2, err := store2.GetReplica(lhsDesc.RangeID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	rhsRepl2, err := store2.GetReplica(rhsDesc.RangeID)
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	// Abandon the two ranges on store2, but do not GC them.
 	mtc.unreplicateRange(lhsDesc.RangeID, 2)
@@ -1755,13 +1766,6 @@ func TestStoreRangeMergeReadoptedBothFollowers(t *testing.T) {
 	_, pErr := client.SendWrapped(ctx, store0.TestSender(), args)
 	if pErr != nil {
 		t.Fatal(pErr)
-	}
-
-	// Attempt to re-add the merged range to store2. The operation should fail
-	// because store2's LHS and RHS replicas intersect the merged range.
-	lhsRepl0, err := store0.GetReplica(lhsDesc.RangeID)
-	if err != nil {
-		t.Fatal(err)
 	}
 
 	addLHSRepl2 := func() error {
@@ -1778,6 +1782,8 @@ func TestStoreRangeMergeReadoptedBothFollowers(t *testing.T) {
 		return nil
 	}
 
+	// Attempt to re-add the merged range to store2. The operation should fail
+	// because store2's LHS and RHS replicas intersect the merged range.
 	err = addLHSRepl2()
 	if exp := "cannot apply snapshot: snapshot intersects existing range"; !testutils.IsError(err, exp) {
 		t.Fatalf("expected %q error, but got %v", exp, err)

